### PR TITLE
Fix #1520: Program returns random data in a function

### LINKED ISSUE #1520: [RPMLint] Program returns random data in a function
When packaging toxcore for openSUSE I encountered a RPMLint error:

     I: Program returns random data in a function
     E: toxcore no-return-in-nonvoid-function ../other/bootstrap_daemon/src/log.c:78, 95

Because the functions ```int level_syslog(LOG_LEVEL level)``` and ```FILE *level_stdout(LOG_LEVEL level)``` do not provide a default return.
I use a simple patch, if you want you can use it:

http://pastie.org/10701281

### PR DESCRIPTION
**Pull request by susnux**
Added default return to non void functions level_syslog and level_stdout.

### CODE REVIEW

File: other/bootstrap_daemon/src/log.c [hunk 1]
```diff
@@ -85,12 +88,13 @@ void log_syslog(LOG_LEVEL level, const char *format, va_list args)
 FILE *level_stdout(LOG_LEVEL level)
 {
     switch (level) {
-        case LOG_LEVEL_INFO:
-            return stdout;
-
```
Comment (nurupo) [inline other/bootstrap_daemon/src/log.c]: Your changes are not similar. In the syslog function you just added a default case, but in stdout you additionally moved `LOG_LEVEL_INFO` case. I would appreciate if you don't move the `LOG_LEVEL_INFO` case from where it originally was.

**Comment by nurupo** [comment]
Also, instead of adding a `default` case, I want you to add a `return` statement outside the `switch`. That way, if new values are added in the enum, a compiler would warn that the `switch` doesn't handle newly added enum cases.

**Comment by susnux** [comment]
Ok I will change that.
But @nurupo gcc will warn also if there is a default, see for example this:
http://pastie.org/10703453
All you have to do is to add a compiler flag: ```-Wswitch-enum```

**Comment by susnux** [comment]
@nurupo Is this ok now?

**Comment by nurupo** [comment]
I'd add a new line before each return, but it looks good as it is.

**Comment by nurupo** [comment]
@GrayHatter yeah, it's good to be merged.

**Comment by nurupo** [comment]
@irungentoo are there any issues with this PR?

**Pull request merged**